pkg/runner/step.go

```diff
@@ -258,10 +259,47 @@ func isContinueOnError(ctx context.Context, expr string, step step, stage stepSt
 	return continueOnError, nil
 }
 
-func mergeIntoMap(target *map[string]string, maps ...map[string]string) {
+func mergeIntoMap(step step, target *map[string]string, maps ...map[string]string) {
+	if rc := step.getRunContext(); rc != nil && rc.JobContainer != nil && rc.JobContainer.IsEnvironmentCaseInsensitive() {
+		mergeIntoMapCaseInsensitive(target, maps...)
+	} else {
+		mergeIntoMapCaseSensitive(target, maps...)
+	}
+}
+
+func mergeIntoMapCaseSensitive(target *map[string]string, maps ...map[string]string) {
 	for _, m := range maps {
 		for k, v := range m {
 			(*target)[k] = v
 		}
 	}
 }
```

Comment:
I think this should work too:
```suggestion
func mergeIntoMapCaseSensitive(target map[string]string, maps ...map[string]string) {
	for _, m := range maps {
		for k, v := range m {
			target[k] = v
		}
	}
}
```